lib/child_process.js

```diff
@@ -50,7 +50,24 @@ exports.fork = function(modulePath /*, args, options*/) {
 
   args = execArgv.concat([modulePath], args);
 
-  if (!Array.isArray(options.stdio)) {
+  if (typeof options.stdio == 'string') {
+    const stdioStringToArray = (option) => {
```

Comment:
Well, the main idea is to avoid the possibility of V8 having to recompile/reoptimize the function. By pulling it out of the function, that can be achieved.